### PR TITLE
Complete dartdocs for rendering.dart

### PR DESCRIPTION
**Pull request by abarth**
All of the public APIs in rendering.dart now have dartdocs.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/rendering/layer.dart [hunk 1]
```diff
@@ -126,10 +129,19 @@ class ChildSceneLayer extends Layer {
     this.sceneToken
   });
 
+  /// Offset from parent in the parent's coordinate system.
   Offset offset;
+
+  /// The number of physical pixels the child should produce for each logical pixel.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/layer.dart]: Is this really physical pixels? e.g. consider if the ChildSceneLayer is inside a transformed layer with a scale and a rotation and a skew.

Comment (abarth) [inline packages/flutter/lib/src/rendering/layer.dart]: Yes, it needs to take that information into account.  The child needs to produce physical pixels so it needs to be informed how much scaling is going to happen by its parents.  (I'm sure we don't compute this accurately yet.)

File: packages/flutter/lib/src/rendering/table.dart [hunk 2]
```diff
@@ -402,6 +476,8 @@ class RenderTable extends RenderBox {
     assert(rows == null || children == null);
     assert(defaultColumnWidth != null);
     _columns = columns ?? (children != null && children.length > 0 ? children.first.length : 0);
+    // If [rows] is non-zero, won't we get into trouble when we call [addRow]
+    // below, which increments _rows for each sublist in [children]?
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/table.dart]: children can't be non-null if children is non-null. One or the other (or both) must be null.

Comment (Hixie) [inline packages/flutter/lib/src/rendering/table.dart]: i guess that should be mentioned in the dartdoc too

File: packages/flutter/lib/src/rendering/table.dart [hunk 3]
```diff
@@ -420,6 +496,11 @@ class RenderTable extends RenderBox {
   // _children.length must be rows * columns
   List<RenderBox> _children = const <RenderBox>[];
 
+  /// The number of vertical alignment lines in this table.
+  ///
+  /// Changing the number of columns will cause the cells in the table to be
+  /// rearranged into the specified number of columns, which is an expensive
+  /// operation.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/table.dart]: It doesn't rearrange the cells. It "just" chops off any cells that were in columns that no longer exist.

It is still expensive though.

File: packages/flutter/lib/src/rendering/table.dart [hunk 4]
```diff
@@ -154,7 +169,14 @@ class FractionColumnWidth extends TableColumnWidth {
...
 /// This is a cheap way to size a column.
 class FlexColumnWidth extends TableColumnWidth {
+  /// Creates a column width based on a fraction of the remaining space once all
+  /// the other columns have been laid out.
+  ///
+  /// The [value] argument must not be null.
   const FlexColumnWidth([this.value = 1.0]);
+
+  /// The reaction of the of the remaining space once all the other columns have
+  /// been laid out that this column shoud occupy.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/table.dart]: should

File: packages/flutter/lib/src/rendering/table.dart [hunk 5]
```diff
@@ -534,6 +639,7 @@ class RenderTable extends RenderBox {
     }
   }
 
+  /// How cells that do not explicitly specify a vertical alignemnt are aligned vertically.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/table.dart]: alignemnt

File: packages/flutter/lib/src/scheduler/binding.dart [hunk 6]
```diff
@@ -343,19 +343,24 @@ abstract class SchedulerBinding extends BindingBase {
...
-  /// [ui.window.scheduleFrame].
+  /// Schedules a new frame using [scheduleFrame] if this object is not
+  /// currently producing a frame.
   ///
-  /// After this is called, the engine will (eventually) call
-  /// [handleBeginFrame]. (This call might be delayed, e.g. if the
-  /// device's screen is turned off it will typically be delayed until
-  /// the screen is on and the application is visible.)
+  /// After this is called, the framework ensures that the end of the
+  /// [handleBeginFrame] function wil (eventually) be reached.
```
Comment (Hixie) [inline packages/flutter/lib/src/scheduler/binding.dart]: will

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
Might be good to discuss the costs of the vertical alignment of cells somewhere. Looks like none of the docs talk about what's cheap and what isn't.

**Pull request merged**